src/write.c

```diff
@@ -585,6 +586,12 @@ static avifResult avifEncoderAddImageInternal(avifEncoder * encoder,
         if (cellImage->yuvFormat == AVIF_PIXEL_FORMAT_NONE) {
             return AVIF_RESULT_NO_YUV_FORMAT_SELECTED;
         }
+
+        // MIAF (ISO 23000-22:2019), Section 7.3.11.4.1:
+        //   All input image of a grid image item shall use the same chroma sampling format.
+        if (cellImage->yuvFormat != firstCell->yuvFormat) {
+            return AVIF_RESULT_INVALID_IMAGE_GRID;
+        }
```

Comment:
It seems better to merge this check into the existing if statement at lines 576-580.

Also, I wonder if we should match the check performed in src/read.c:

```
        if ((tile->image->width != firstTile->image->width) || (tile->image->height != firstTile->image->height) ||
            (tile->image->depth != firstTile->image->depth) || (tile->image->yuvFormat != firstTile->image->yuvFormat) ||
            (tile->image->yuvRange != firstTile->image->yuvRange) || (uvPresent != firstTileUVPresent) ||
            (tile->image->colorPrimaries != firstTile->image->colorPrimaries) ||
            (tile->image->transferCharacteristics != firstTile->image->transferCharacteristics) ||
            (tile->image->matrixCoefficients != firstTile->image->matrixCoefficients) ||
            (tile->image->alphaRange != firstTile->image->alphaRange)) {
            avifDiagnosticsPrintf(data->diag, "Grid image contains mismatched tiles");
            return AVIF_FALSE;
        }
```